server/src/resolvers/Source.ts

```diff
@@ -183,3 +185,77 @@ export const deleteSource: FieldResolver<'Mutation', 'deleteSource'> = async (
   )
   return ctx.prisma.source.delete({ where: { id: sourceId } })
 }
+
+export const usedConceptMaps: FieldResolver<
+  'Query',
+  'usedConceptMapIds'
+> = async (_, { sourceId }, ctx) => {
+  const source = await ctx.prisma.source.findOne({
+    where: { id: sourceId },
+    include: { template: true },
+  })
+  if (!source) {
+    throw new Error(`source ${sourceId} does not exist`)
+  }
+
+  const sourceWithMapIds = await ctx.prisma.source.findOne({
+    where: { id: sourceId },
+    include: {
+      resources: {
+        include: {
+          attributes: {
+            include: {
+              inputs: true,
+            },
+          },
+        },
+      },
+    },
+  })
+  const resources = sourceWithMapIds!.resources as ResourceWithAttributes[]
+
+  const reduceattributes = (
+    acc: string[],
+    curAttribute: AttributeWithInputs,
+  ) => [
+    ...acc,
+    ...(curAttribute.inputs
+      .map(input => input.conceptMapId)
+      .filter(Boolean) as string[]),
+  ]
+  const reduceResources = (
+    acc: string[],
+    curResource: ResourceWithAttributes,
+  ) => [...acc, ...curResource.attributes.reduce(reduceattributes, [])]
+
+  return resources.reduce(reduceResources, [])
+}
+
+export const usedProfiles: FieldResolver<'Query', 'usedProfileIds'> = async (
+  _,
+  { sourceId },
+  ctx,
+) => {
+  const source = await ctx.prisma.source.findOne({
+    where: { id: sourceId },
+    include: { template: true },
+  })
+  if (!source) {
+    throw new Error(`source ${sourceId} does not exist`)
+  }
+
+  const resources = await ctx.prisma.resource.findMany({
+    where: { source: { id: sourceId } },
+  })
+
+  const definitions = await Promise.all(
+    resources.map(r => getDefinition(r.definitionId)),
+  )
+
+  const profileIds = definitions
+    .filter(def => !!def && def.meta.id !== def.meta.type)
```

Comment:
I think it's be best to use the `definition.meta.derivation` which equals to "specialization" if it's a profile and "constraint" if it's a base resource. also, can you explain the case where def is `null` (why do you need !!def ?)

Reply:
I think the !! was there because of typing (the find above returns a list of defs | undefined)

Reply:
Agreed for the profile check, I'll change that